text/0419-cdk-launchpads.md

```diff
@@ -0,0 +1,632 @@
+In that scenario, the Launchpad is basically a CDK application, and you execute the Launchpads CLI commands,
+like `cdk-lpads update`, inside of it, similarly like you execute CDK CLI commands inside your CDK application.
+
+A CDK application for a Launchpad has a very similar structure to a "regular" CDK application,
+but the classes used are slightly different (the example is in TypeScript):
+
+```ts
+import * as lpads from '@aws-cdk/cdk-launchpads';
+
+const launchpad = new lpads.Launchpad();
+new lpads.StandardCdkBootstrap(launchpad, 'CdkBootstrap', {
+  // options to customize the CDK bootstrap can be passed here...
+});
+
+// additional things can be changed in the Launchpad here...
+
+launchpad.synth();
+```
+
+This CDK application will synthesize to the JSON configuration mentioned above,
+similarly like "regular" CDK applications synthesize to CloudFormation templates.
+
+We will show details of customizing the Launchpad using CDK code in each of the sections below.
+
+### Launchpad features
+
+#### Compliance
+
+Compliance rules allow administrators to restrict what kind of AWS resources can be deployed in a given environment.
+They can be used both for security purposes,
+but also for things like checking whether a given application conforms to the AWS Well-Architected principles.
+
+Launchpads allow defining compliance rules using the
+[CFN Guard 2.0 tool](https://github.com/aws-cloudformation/cloudformation-guard).
+Those compliance rules are then deployed as
+[CloudFormation Hooks](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks.html),
+which means it's impossible to deploy a CDK application into a given environment that violates any compliance rule defined for it.
+
+When importing the Launchpad into your CDK app (see below),
+the Compliance rules get imported as well,
+meaning they will be evaluated at `synth` time in addition to at deploy time,
+to provide developers the quickest possible feedback loop on whether their apps are compliant or not.
+
+Compliance rules can be defined in JSON/YAML directly:
+
+```json
+{
+  "launchpad": {
+    "compliance": {
+      "cfn-guard": [
+        {
+          "common": [
+            "let s3_buckets = Resources.[ Type == /S3::Bucket/ ]"
+          ],
+          "rules": {
+            "s3_bucket_name_encryption_check": {
+              "$when": "%s3_buckets !empty",
+              "%s3_buckets": {
+                "Properties": {
+                  "BucketName": "== /^MyCompanyPrefix/",
+                  "BucketEncryption.ServerSideEncryptionConfiguration[]": {
+                    "ServerSideEncryptionByDefault.SSEAlgorithm": "IN ['aws:KMS']"
+                  }
+                }
+              }
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+Or you can reference a file on disk that contains the Guard rules:
+
+```json
+{
+  "launchpad": {
+    "compliance": {
+      "cfn-guard": [
```

Comment:
@skinny85 are you sure that CFN Guard accepts multiple values? Or are you planning to call CFN Guard multiple times and merge the results if so?

Reply:
\+ referencing discussion we started which became outdated after the last commit:
https://github.com/aws/aws-cdk-rfcs/pull/421#discussion_r861462006

Reply:
Yes, I think we can easily run the command multiple times, and merge the results.